nemo/collections/tts/models/two_stages.py

```diff
@@ -0,0 +1,225 @@
+
+from nemo.collections.tts.helpers.helpers import griffin_lim
+from nemo.collections.tts.models.base import LinVocoder, MelToSpec, Vocoder
+from nemo.core.classes.common import PretrainedModelInfo
+from nemo.core.neural_types.elements import (
+    AudioSignal,
+    LengthsType,
+    MelSpectrogramType,
+    NormalDistributionSamplesType,
+    SpectrogramType,
+    VoidType,
+)
+from nemo.core.neural_types.neural_type import NeuralType
+from nemo.utils import logging
+
+
+class OperationMode(Enum):
+    """Training or Inference (Evaluation) mode"""
+
+    training = 0
+    validation = 1
+    infer = 2
+
+
+class MelPsuedoInverseModel(MelToSpec):
+    def __init__(self, cfg: DictConfig):
+        if isinstance(cfg, dict):
+            cfg = OmegaConf.create(cfg)
+
+        super().__init__(cfg=cfg)
+
+        sampling_rate = self._cfg['sampling_rate']
+        n_fft = self._cfg['n_fft']
+        mel_fmin = self._cfg['mel_fmin']
+        mel_fmax = self._cfg['mel_fmax']
+        mel_freq = self._cfg['mel_freq']
+
+        melinv = librosa.filters.mel(sr=sampling_rate, n_fft=n_fft, fmin=mel_fmin, fmax=mel_fmax, n_mels=mel_freq)
+        self.mel_pseudo_inverse = torch.tensor(melinv, dtype=torch.float)
+
+    def convert_mel_spectrogram_to_linear(self, mel):
+        lin_spec = torch.tensordot(mel, self.mel_pseudo_inverse, dims=[[1], [0]])
+        lin_spec = lin_spec.permute(0, 2, 1)
+        return lin_spec
+
+    def setup_training_data(self, cfg):
+        pass
+
+    def setup_validation_data(self, cfg):
+        pass
+
+    def cuda(self, *args, **kwargs):
+        self.mel_pseudo_inverse = self.mel_pseudo_inverse.cuda(*args, **kwargs)
+        return self
+
+
+class GriffinLimModel(LinVocoder):
+    def __init__(self, cfg: DictConfig):
+        if isinstance(cfg, dict):
+            cfg = OmegaConf.create(cfg)
+
+        super().__init__(cfg=cfg)
+
+        self.n_iters = self._cfg['n_iters']
+        self.n_fft = self._cfg['n_fft']
+        self.l_hop = self._cfg['l_hop']
+
+    def convert_linear_spectrogram_to_audio(self, spec, Ts=None):
+        batch_size = spec.shape[0]
+
+        T_max = spec.shape[2]
+        if Ts is None:
+            Ts = [T_max] * batch_size
+
+        max_size = (max(Ts) - 1) * self.l_hop
+        audios = torch.zeros(batch_size, max_size)
+        # Lazy GL implementation. Could be improved by moving to pytorch.
+        for i in range(batch_size):
+            audio = griffin_lim(spec[i, :, 0 : Ts[i]].cpu().numpy() ** 1.2, n_iters=self.n_iters, n_fft=self.n_fft)
+            my_len = audio.shape[0]
+            audios[i, 0:my_len] = torch.from_numpy(audio)
+
+        return audios
+
+    def setup_training_data(self, cfg):
+        pass
+
+    def setup_validation_data(self, cfg):
+        pass
+
+    def cuda(self, *args, **kwargs):
+        return self
+
+
+@dataclass
+class TwoStagesConfig:
+    mel2spec: Dict[Any, Any] = MISSING
+    linvocoder: Dict[Any, Any] = MISSING
+
+
+class TwoStagesModel(Vocoder):
```

Comment:
I feel like this is a general enough model that we should move to the base.py file.